ui/apps/platform/src/Containers/Collections/CollectionFormModal.tsx

```diff
@@ -0,0 +1,107 @@
+        content = (
+            <>
+                {error.message}
+                {/* TODO - Handle UI for network errors */}
+            </>
+        );
+    } else if (loading) {
+        content = <>{/* TODO - Handle UI for loading state */}</>;
+    } else if (data) {
+        content = (
+            <Modal
+                isOpen
+                onClose={onClose}
+                aria-label={`View ${data.collection.name}`}
+                width="90vw"
+                hasNoBodyWrapper
+                header={
+                    <Flex className="pf-u-pb-md" alignItems={{ default: 'alignItemsCenter' }}>
+                        <FlexItem grow={{ default: 'grow' }}>
+                            <Title headingLevel="h2">{data.collection.name}</Title>
+                        </FlexItem>
+                        {hasWriteAccessForCollections && (
+                            <Button
+                                variant="link"
+                                component="a"
+                                href={`${collectionsBasePath}/${collectionId}?action=edit`}
+                                target="_blank"
+                                rel="noopener noreferrer"
+                                icon={<ExternalLinkAltIcon />}
+                            >
+                                Edit collection
+                            </Button>
+                        )}
+                        {isDrawerOpen ? (
+                            <Button variant="secondary" onClick={closeDrawer}>
+                                Hide results
+                            </Button>
+                        ) : (
+                            <Button variant="secondary" onClick={openDrawer}>
+                                Preview results
+                            </Button>
+                        )}
+                        <Divider orientation={{ default: 'vertical' }} component="div" />
+                    </Flex>
+                }
+            >
+                <Divider component="div" />
+                <CollectionFormDrawer
+                    // We do not want to present the user with options to change the collection when in this modal
+                    hasWriteAccessForCollections={false}
+                    action={{
+                        type: 'view',
+                        collectionId,
+                    }}
+                    collectionData={data}
+                    isInlineDrawer={isLargeScreen}
+                    isDrawerOpen={isDrawerOpen}
+                    toggleDrawer={toggleDrawer}
+                    // Since the form cannot be submitted, stub this out with an empty promise
+                    onSubmit={() => Promise.resolve()}
+                    appendTableLinkAction={(id) => {
+                        const url = `${window.location.origin}${collectionsBasePath}/${id}`;
+                        window.open(url, '_blank', 'noopener noreferrer');
+                    }}
```

Comment:
First a simple question about naming. Is there any other use of `appendTableLinkAction` other than as value for `onItemClick` prop? My mind pauses to connect that thought with append in the prop name here.

Reply:
Second, a question about a potential alternative which I feel worth to explore without necessarily recommending. In this case, the result looks like a link, but you do not see address at lower left of browser window if you point at it, because `onClick` prop:

```tsx
<Button
    variant="link"
    className="pf-u-pl-0"
    isInline
    onClick={() => onItemClick(id)}
>
```

I am interested in any pro and con for accessibility from PatternFly team when genuine link has `onClick` handler instead of `href` prop, like the following:

https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/MitreAttackVectors/MitreAttackLink.tsx#L12-L23

If, and only if, feedback suggested that there is benefit to outweigh the cost, would it make sense to change prop value from callback function to `Button` element which encapsulates `href` or `onClick` prop.

Reply:
I agree, the `appendTableLinkAction` is a bit of an artifact from some earlier plans that have evolved with design changes. Additionally, the read-only table that replaces the CollectionAttacher has some duplication with the cells that are rendered so I should be able to consolidate if we are passing the components down.

TBH I don't think there are any pros to using an onclick/window.open in this case instead of a link button. The main pro was to avoid some indirection and prop drilling, but since we are passing the callback just as deeply as we would be the components, I don't think it outweighs the cons of no link preview and programmatic navigation in this case.

Reply:
Since we are passing components instead of callbacks, I was able to add the "external link" icon to the table cells, much like the ones in the header. It better conveys what the link will do, but do you think this is too much?

![image](https://user-images.githubusercontent.com/1292638/201691386-00abaec5-c841-4987-b99b-91b713656406.png)

Reply:
Because my mind uses ”one of these things is not like the others” principle, I would notice that **Edit collection** has icon but collection names do not. Therefore, I think it is ”just right” to mix metaphors.

Two observations which you are free to choose according to house special for today: take it, or leave it ;-)
* Although icon at the left of collection name seems superior for alignment in the table, icon at the right feels more fluent for **Edit collection** link.
* The first picture under **Description** both wraps name and truncates description of the first collection. Although every additional style risks trouble in a responsive layout, what do you think are pro and con of no wrap for name?

Reply:
I'm neutral on the first point, so a positive and a neutral == positive, I'll move the icon to the right.

For the second point, I like the idea but this leaves us open to very long user input breaking the layout of the page due to the nowrap. I've felt the table layouts need a second look though, so I'll add this to my list of follow ups to take a closer look at.